### ISSUE TITLE
Missing dependency `dill` from `setup.py`.

### ISSUE THREAD

**Issue by UriyaHarpeness** [jpmml/sklearn2pmml#389]
Background
---
In the following commit: [Added 'dump_flavour' parameter](https://github.com/jpmml/sklearn2pmml/commit/2a12957d7c514e76fa2a1421b33eec9c299a8a4b).
In `sklearn2pmml/__init__.py` this line was added `import dill`, and while it was added to `requirements.txt`, it was not added to `setup.py`, where the dependencies of the package are resolved for consumers.

Traceback
---
```python
    import sklearn2pmml
/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/sklearn2pmml/__init__.py:19: in <module>
    import dill
E   ModuleNotFoundError: No module named 'dill'
```

Solution
---
Either add the `dill` dependency to `setup.py`, or protect the use of diff with `except ImportError` and fallback.

**Comment by UriyaHarpeness** [comment]
Tagging you @vruusmann. If you want I can submit a PR that adds `dill` to `setup.py`.

**Comment by vruusmann** [comment]
The support for `dill` and `joblib` flavours of the Pickle data format is a "systemically important" decision. So, the `dill` package requirement will go straight into `setup.py`.

Right now, dill was added, because it is able to persist fitted H2O.ai estimators (joblib isn't). In the future, it might come in handy when trying to persist `sklearn.preprocessing.FunctionTransformer` transformers that contain a reference to user-defined functions or lambdas.

**Comment by UriyaHarpeness** [comment]
@vruusmann I'm not sure if you're aware of it, but currently this fix is still not released to PyPI. So as of now, installing and importing `sklearn2pmml` on its own fails on `ImportError` of `dill`.

**Comment by vruusmann** [comment]
> currently this fix is still not released to PyPI.

This bug was caused by/reported right after the 0.95.0 release.

I'm not quite sure if this bug fix is wort doing a dedicated PyPI release (say, 0.95.1) or now. Right now I was waiting to see if there is something more to put into this new release.

@UriyaHarpeness  WDYT,  is a dedicated release necessary or not?

**Comment by UriyaHarpeness** [comment]
According to semantic versioning, this is exactly the case to advance the version with a patch to 0.95.1.
I myself have several CIs that are having problems because of this issue, and I'd rather not limit the package's version as to not miss improvements.
So to sum up I'm all for it!

**Comment by vruusmann** [comment]
I've prepared and released SkLearn2PMML version 0.95.1 to PyPI:
https://pypi.org/project/sklearn2pmml/0.95.1/

The next CI run should get back to normal all by itself now!

**Comment by UriyaHarpeness** [comment]
Works! Thanks a lot @vruusmann, I appreciate your work.
